File: facade/php/Facade.php
<?php

// Subsystem: CPU
class CPU {
    public function freeze(): string {
        return "CPU: Freezing processor";
    }

    public function jump(string $position): string {
        return "CPU: Jumping to position {$position}";
    }

    public function execute(): string {
        return "CPU: Executing instructions";
    }
}

// Subsystem: Memory
class Memory {
    public function load(string $position, string $data): string {
        return "Memory: Loading '{$data}' at position {$position}";
    }
}

// Subsystem: HardDrive
class HardDrive {
    public function read(string $sector, string $size): string {
        return "HardDrive: Reading {$size} bytes from sector {$sector}";
    }
}

// Facade
class ComputerFacade {
    private $cpu;
    private $memory;
    private $hardDrive;

    public function __construct() {
        $this->cpu = new CPU();
        $this->memory = new Memory();
        $this->hardDrive = new HardDrive();
    }

    public function start(): array {
        $results = [];
        $results[] = "Starting Computer...";
        $results[] = $this->cpu->freeze();
        $results[] = $this->memory->load("0x00", "BOOT_ADDRESS");
        $results[] = $this->hardDrive->read("0", "1024");
        $results[] = $this->cpu->jump("0x00");
        $results[] = $this->cpu->execute();
        $results[] = "Computer started successfully!";
        return $results;
    }
}

// Usage
$computer = new ComputerFacade();
$results = $computer->start();
foreach ($results as $result) {
    echo $result . "\n";
}

File: patterns/facade/python/Facade.py
from typing import List

# Subsystem: CPU
class CPU:
    def freeze(self) -> str:
        return "CPU: Freezing processor"

    def jump(self, position: str) -> str:
        return f"CPU: Jumping to position {position}"

    def execute(self) -> str:
        return "CPU: Executing instructions"

# Subsystem: Memory
class Memory:
    def load(self, position: str, data: str) -> str:
        return f"Memory: Loading '{data}' at position {position}"

# Subsystem: HardDrive
class HardDrive:
    def read(self, sector: str, size: str) -> str:
        return f"HardDrive: Reading {size} bytes from sector {sector}"

# Facade
class ComputerFacade:
    def __init__(self):
        self._cpu = CPU()
        self._memory = Memory()
        self._hard_drive = HardDrive()

    def start(self) -> List[str]:
        results = []
        results.append("Starting Computer...")
        results.append(self._cpu.freeze())
        results.append(self._memory.load("0x00", "BOOT_ADDRESS"))
        results.append(self._hard_drive.read("0", "1024"))
        results.append(self._cpu.jump("0x00"))
        results.append(self._cpu.execute())
        results.append("Computer started successfully!")
        return results


if __name__ == "__main__":
    computer = ComputerFacade()
    for result in computer.start():
        print(result)
